File: 1.c
#include <cstdio.h>

// write a C version when I have time

class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            vector<int> tmp = nums;
            sort(nums.begin(),nums.end());
            for (int i = 0; i < nums.size(); ++i){
                for (int j = i+1; j < nums.size(); ++j){
                    if (target == nums[i] + nums[j]){
                        vector<int> ret ;
                        bool a,b;
                        a = b = false;
                        for(int k = 0; k < nums.size(); ++k){
                            if (nums[i] == tmp[k] && !a){
                                ret.push_back(k+1);
                                a = true;
                            } else if (nums[j] == tmp[k] && !b){
                                ret.push_back(k+1);
                                b = true;
                            }
                            if(a&&b) return ret;
                        }
                    }
                }
            }
        }
};

File: 1/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  for ( var i = 1; i < nums.length; ++i ) {
    for ( var j = 0; j < i; ++j) {
      if (nums[i] + nums[j] == target) {
        return [i, j];
      }
    }
  }
};

// console.log( twoSum([2, 7, 11, 15], 9) );
// console.log( twoSum([0,4,3,0], 0) );
// console.log( twoSum([3,2,4], 6) );
